server/workers/shareScrollWorker.ts

```diff
@@ -0,0 +1,548 @@
+        '8px Nokia Cellphone FC',
+        ''
+      ),
+      left: 200,
+      top: 28,
+    });
+    composites.push({
+      input: await textToPng(
+        statText('All-time Clicks', allTimeClicks),
+        '8px Nokia Cellphone FC',
+        ''
+      ),
+      left: 200,
+      top: 40,
+    });
+
+    console.log(
+      `Banner stats generated in ${(performance.now() - startTime).toFixed(2)}ms`
+    );
+    const bannerBuffer = await compositeImage
+      .composite(composites)
+      .png()
+      .toBuffer();
+    return {
+      bannerBuffer,
+      bannerHeight: BANNER_HEIGHT,
+      bannerWidth: BANNER_WIDTH,
+    };
+  } catch (error) {
+    console.error('Error in getBannerBase:', error);
+    throw error;
+  }
+}
+
+//
+
+async function getDragonStrip(dragonIds: string[]) {
+  // it's fine for there to be no growing things,
+  // it just means the carousel will be empty.
+  if (dragonIds.length === 0) {
+    return {
+      stripBuffer: null,
+      stripWidth: 0,
+      stripHeight: 0,
+    };
+  }
+
+  try {
+    const startTime = performance.now();
+    console.log('Generating growing images...');
+    const dragonBuffers = await Promise.all(
+      dragonIds.map(async (dragonId) => {
+        const response = await fetch(
+          `https://dragcave.net/image/${dragonId}.gif`
+        );
+        if (!response.ok) {
+          throw new Error(
+            `Failed to fetch dragon image ${dragonId}: ${response.statusText}`
+          );
+        }
+        const arrayBuffer = await response.arrayBuffer();
+        return Buffer.from(arrayBuffer);
+      })
+    );
+
+    const dragonImages = dragonBuffers.map((buffer) => sharp(buffer));
+
+    const dragonMetadatas = await Promise.all(
+      dragonImages.map((img) => img.metadata())
+    );
+
+    const totalWidth = dragonMetadatas.reduce(
+      (acc, curr) => acc + (curr.width ?? 0) + 1,
+      0
+    );
+
+    const HEIGHT = 50;
+    // height will be solid, no point in having it be flexible.
+    // plus, dragons too tall will just be cropped
+
+    let compositeImage = createEmptyFrame(totalWidth, HEIGHT);
+
+    const composites: sharp.OverlayOptions[] = [];
+    let xOffset = 0;
+
+    for (let i = 0; i < dragonImages.length; i++) {
+      const dragonImage = dragonImages[i];
+      const metadata = dragonMetadatas[i];
+
+      composites.push({
+        input: await dragonImage.toBuffer(),
+        top: HEIGHT - (metadata.height ?? 0),
+        left: xOffset,
+      });
+
+      xOffset += (metadata.width ?? 0) + 1;
+    }
+
+    const stripBuffer = await compositeImage
+      .composite(composites)
+      .png()
+      .toBuffer();
+
+    console.log(
+      `Growing images generated in ${(performance.now() - startTime).toFixed(0)}ms`
+    );
+    return {
+      stripBuffer,
+      stripWidth: totalWidth,
+      stripHeight: HEIGHT,
+    };
+  } catch (error) {
+    console.error('Error in getDragonStrip:', error);
+    throw error;
+  }
```

Comment:
I'm not sure this serves any purpose - catching an error and then rethrowing it.

Reply:
Blocks [like these](https://github.com/edenchazard/garden-of-eden/blob/3df5fe0331dfc493a3eec4d6083e1c127a31ffb9/server/workers/shareScrollWorker.ts#L224C3-L227C4) were present before I worked on it. In its defense I do feel telling us where in the generation process an error occurred can be convenient in case the stack trace alone isn't helpful.